libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/EHorizon.kt

```diff
@@ -122,26 +98,32 @@ class EHorizon private constructor(
             ")"
     }
 
-    /**
-     * Builder for [EHorizon].
-     */
-    class Builder {
-
-        private var start: Edge = Edge.Builder().build()
-
-        /**
-         * Defines the start [Edge]
-         */
-        fun start(start: Edge): Builder =
-            apply { this.start = start }
-
-        /**
-         * Build the [EHorizon]
-         */
-        fun build(): EHorizon {
-            return EHorizon(
-                start = start
-            )
+    private fun mpp(start: Edge): List<List<Edge>> {
```

Comment:
We should add unit tests here.

Test cases:

1)

1-2-3-4-5

2)

1-2-3-4-5. (starting 2) -> 2-3-4-5- (edited) 

3)

0 0
 |. |
1-2-3-4-0
 |   |. |
 0  0   \5-6-7 
->    1-2-3-4,   1-2-3-5-6-7

4)

from 2 ->  2-3-4, 2-3-5-6-7

5)

from 4 -> 4

6)

from 3 -> 3-4 3-5-6-7